### PR TITLE
[VL] Support Celeborn integration testing in Velox CI

### PR DESCRIPTION
**Pull request by kerwin-zk**
## What changes were proposed in this pull request?
Support Celeborn Integration Testing in Velox CI

### CODE REVIEW

**Comment by kerwin-zk** [comment]
@zhztheplayer @jinchengchenghh @zhouyuan cc

File: .github/workflows/velox_be.yml [hunk 1]
```diff
@@ -103,6 +103,22 @@ jobs:
             --backend-type=velox --benchmark-type=h --fixed-width-as-double --disable-aqe --off-heap-size=20g -s=1.0 --cpus=16 --iterations=1 \
           && java -Xmx5G -XX:ErrorFile=/var/log/java/hs_err_pid%p.log -cp target/gluten-it-1.0-SNAPSHOT-jar-with-dependencies.jar io.glutenproject.integration.tpc.Tpc \
             --backend-type=velox --benchmark-type=ds --fixed-width-as-double --off-heap-size=8g -s=0.1 --cpus=16 --iterations=1'
+      - name: Create Celeborn Cluster
```
Comment (jinchengchenghh) [inline .github/workflows/velox_be.yml:106]: Can Celeborn work on Ubuntu22.04? Now the Unbun20.04 CI time is too long. https://stackoverflow.com/questions/51613897/how-can-i-get-ubuntu-version-using-java

Comment (kerwin-zk) [reply .github/workflows/velox_be.yml:106]: @jinchengchenghh Celeborn works on Ubuntu22.04. The long time of Ubuntu20.04 CI is mainly due to the newly added TPCDS and TPCH tests of Spark3.2 and Spark3.3 using Celeborn. About 20 minutes longer than before.

Comment (jinchengchenghh) [reply .github/workflows/velox_be.yml:106]: Maybe we can move these new tests to Velox backend / velox-backend-ubuntu2204-test

Comment (kerwin-zk) [reply .github/workflows/velox_be.yml:106]: @jinchengchenghh  I have added tests for both Ubuntu 20.04 and Ubuntu 22.04. If Ubuntu 22.04 is enough, I can remove Ubuntu 20.04.  @zhztheplayer @zhouyuan cc

File: tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/TpcSuite.scala [hunk 2]
```diff
@@ -76,6 +77,16 @@ abstract class TpcSuite(
...
 
+  if (enableCeleborn) {
+    sessionSwitcher.defaultConf().set("spark.shuffle.manager",
+      "org.apache.spark.shuffle.celeborn.CelebornShuffleManager")
+    sessionSwitcher.defaultConf().set("spark.celeborn.shuffle.writer", "hash")
+    sessionSwitcher.defaultConf().set("spark.celeborn.push.replicate.enabled", "false")
+    sessionSwitcher.defaultConf().set("spark.shuffle.service.enabled", "false")
+    sessionSwitcher.defaultConf().set("spark.sql.adaptive.localShuffleReader.enabled", "false")
+    sessionSwitcher.defaultConf().set("spark.dynamicAllocation.enabled", "false")
+  }
```
Comment (zhztheplayer) [inline tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/TpcSuite.scala:88]: `sessionSwitcher.defaultConf()` will be applied to both baseline session (Vanilla Spark) and test session (Gluten). Do these config options make Celeborn work on Vanilla Spark correctly? Did you check on that?

File: tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/ds/TpcdsSuite.scala [hunk 3]
```diff
@@ -24,9 +25,9 @@ class TpcdsSuite(
   val shufflePartitions: Int,
   val minimumScanPartitions: Boolean) extends TpcSuite(
   actions, testConf, baselineConf, fixedWidthAsDouble,
-  logLevel, errorOnMemLeak, enableUi, enableHsUi, hsUiPort, cpus,
-  offHeapSize, disableAqe, disableBhj, disableWscg, shufflePartitions,
-  minimumScanPartitions) {
+  logLevel, errorOnMemLeak, enableUi, enableHsUi, enableCeleborn,
+  hsUiPort, cpus, offHeapSize, disableAqe, disableBhj, disableWscg,
+  shufflePartitions, minimumScanPartitions) {
```
Comment (zhztheplayer) [inline tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/ds/TpcdsSuite.scala:30]: Let's make `enableHsUi` and `hsUiPort` conjective.

File: tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/TpcSuite.scala (hunk 2 above)
Comment (kerwin-zk) [reply tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/TpcSuite.scala]: @zhztheplayer Celeborn configurations have been added to test session.

File: tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/ds/TpcdsSuite.scala (hunk 3 above)
Comment (kerwin-zk) [reply tools/gluten-it/src/main/scala/io/glutenproject/integration/tpc/ds/TpcdsSuite.scala]: @zhztheplayer done.

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by kerwin-zk** [comment]
@zhztheplayer @zhouyuan cc

**Review by zhztheplayer — APPROVED** [review]

**Comment by zhztheplayer** [comment]
Thanks @kerwin-zk !

File: .github/workflows/velox_be.yml (hunk 1 above)
Comment (zhztheplayer) [reply .github/workflows/velox_be.yml]: > @jinchengchenghh I have added tests for both Ubuntu 20.04 and Ubuntu 22.04. If Ubuntu 22.04 is enough, I can remove Ubuntu 20.04. @zhztheplayer @zhouyuan cc

I am changing to make celeborn CI use 22.04 only in https://github.com/oap-project/gluten/pull/1294. Please feel free to comment if having any issue.

**Pull request merged**